### ISSUE TITLE
Let's not go to innerHTML, tis a silly place.

### ISSUE THREAD

**Issue by benshayden** [catapult-project/catapult#2829]
There are a few places where production code uses innerHTML. Let's use DOM methods instead.

 * [tracing/base/base.html](https://github.com/catapult-project/catapult/blob/9be0433d2f769c91b70c4851f1deb8f3f8682412/tracing/tracing/base/base.html#L122)
 * [tracing/ui/analysis/single_object_instance_sub_view.html](https://github.com/catapult-project/catapult/blob/ca7ccf52377f585330be2182bb4dedb50e587e57/tracing/tracing/ui/analysis/single_object_instance_sub_view.html#L109)
 * [tracing/ui/base/radio_picker.html](https://github.com/catapult-project/catapult/blob/bcb075c8a1e3456bca2d06badb47e819d98320e5/tracing/tracing/ui/base/radio_picker.html#L76)
 * [tracing/ui/base/dom_helpers.html](https://github.com/catapult-project/catapult/blob/b04501d22d696cc7325e4471bee268c0b3f8f85b/tracing/tracing/ui/base/dom_helpers.html#L74)
 * [tracing/ui/extras/drive/index.html](https://github.com/catapult-project/catapult/blob/b04501d22d696cc7325e4471bee268c0b3f8f85b/tracing/tracing/ui/extras/drive/index.html#L320)
 * [telemetry/support/html_output/results-template.html](https://github.com/catapult-project/catapult/blob/a8a6fb7ea065c2df8ae129ad05ac80ba6ae7cc3a/telemetry/support/html_output/results-template.html#L1204)
 * [dashboard/templates/edit_sheriffs.html](https://github.com/catapult-project/catapult/blob/cc2f04068f9747880001ebf797cc2a33abab3179/dashboard/dashboard/templates/edit_sheriffs.html#L91)
 * [dashboard/elements/quick-log.html](https://github.com/catapult-project/catapult/blob/b04501d22d696cc7325e4471bee268c0b3f8f85b/dashboard/dashboard/elements/quick-log.html#L317)
 * [dashboard/elements/test-picker.html](https://github.com/catapult-project/catapult/blob/b04501d22d696cc7325e4471bee268c0b3f8f85b/dashboard/dashboard/elements/test-picker.html#L277)
 * [dashboard/static/debug_alert.html](https://github.com/catapult-project/catapult/blob/b04501d22d696cc7325e4471bee268c0b3f8f85b/dashboard/dashboard/static/debug_alert.html#L167)
 * [dashboard/elements/autocomplete-box.html](https://github.com/catapult-project/catapult/blob/b04501d22d696cc7325e4471bee268c0b3f8f85b/dashboard/dashboard/elements/autocomplete-box.html#L485)

@nedn , I noticed this when I was reading radio-picker while reviewing lpy's CL. Can I ask you to fix at least that one? :-)

**Comment by nedn** [comment]
Sure. Can you explain why DOM methods is better than innerHTML?

**Comment by anniesullie** [comment]
Why not use polymer templates instead of DOM methods or innerHTML for dom generation? @zeptonaut @Apeliotes @eakuefner @natduca

**Comment by benshayden** [comment]
innerHTML is a bit slower than dom methods in Chrome, it can be an XSS security risk, and it can be difficult to read and reason about. DOM methods are OOP like the rest of our code, whereas innerHTML invokes parsing magic that just doesn't really fit with OOP.

Yes, when possible, polymer templates are preferable to DOM methods.

**Comment by zeptonaut** [comment]
I agree with everything that's been said on this bug so far :-)

... but I also have no time to take on fixing these :-(

**Comment by eakuefner** [comment]
Ben, do you want to

1. Give this bug a clearer title
2. Expand the directions a little bit in the description, and
3. Add GoodFirstBug?

**Comment by Dimitrees** [comment]
Hello, I am new to both your project and working with GitHub in general. Can I work on resolving this issue since its labeled as Good First Bug? 

Thanks

**Comment by anniesullie** [comment]
@sfrankian: are you still working on this?

**Comment by sfrankian** [comment]
@anniesullie no, I am not working on this right now and have unassigned myself

**Comment by anniesullie** [comment]
@Dimitrees: Looks like this bug is available! First step is to review the [Contributor's guide](https://github.com/catapult-project/catapult/blob/master/CONTRIBUTING.md). Next step would be to pick a file from the list and update it. You can send CLs to eakuefner@chromium.org.

**Comment by krishamehta** [comment]
Can I help ?

**Comment by kman2121** [comment]
I can take this one on.

**Comment by fridamat** [comment]
We are a group of five computer science students who have been given 20-30 hours each to spend on an issue in an open source project. We are interested in contributing to your project, and this specific issue to start with. We will follow your guide for how to contribute and hope that we can be assigned to and work on this issue. Do all files still need to be updated? In that case, I guess we start from the top.
